pkgs/build-support/buildenv/builder.pl

```diff
@@ -14,14 +14,44 @@
 
 my @pathsToLink = split ' ', $ENV{"pathsToLink"};
 
-sub isInPathsToLink {
+# Normalize paths in pathsToLink. Normalized paths do not have duplicated
+# or trailing slashes and have a leading slash except for the root which is
+# an empty string. E.g.: "", "/foo", "/foo/bar".
+foreach my $pathToLink (@pathsToLink) {
+    my @parts = split(/\//, $pathToLink);
+    @parts = grep { $_ ne '' } @parts;
+    $pathToLink = join "", map { "/" . $_ } @parts;
+}
+
+# Determine if two paths are related, that is they are equal or one is below
+# the other. Both paths must be normalized as done above for pathsToLink
+# (note that $relName in findFiles is normalized due to its construction).
+# Examples: ("/a/b", "") -> 1, ("/a/b", "/a") -> 1, ("/a/b", "/a/b") -> 1,
+# ("/a/b", "/a/b/c") -> 1, ("/a/b", "/c") -> 0, ("/a/b", "/c/d/e") -> 0.
+sub pathsAreRelated {
+    my ($path1, $path2) = @_;
+    my $len1 = length($path1);
+    my $len2 = length($path2);
+    if ($len1 == $len2) {
+        return $path1 eq $path2;
+    } elsif ($len1 < $len2) {
+        return $path1 eq substr($path2, 0, $len1) && substr($path2, $len1, 1) eq "/";
+    } else {
+        return $path2 eq substr($path1, 0, $len2) && substr($path1, $len2, 1) eq "/";
```

Comment:
I looked at this again and there is one corner case left: `('/a', '/') -> 0`

```suggestion
        return $path1 eq substr($path2, 0, $len1) && substr($path2, $len1 == 1 ? 0 : $len1, 1) eq "/";
    } else {
        return $path2 eq substr($path1, 0, $len2) && substr($path1, $len2 == 1 ? 0 : $len2, 1) eq "/";
```